### ISSUE TITLE
VB6 - Code Explorer Add should support templates

### ISSUE THREAD

**Issue by mansellan** [rubberduck-vba/Rubberduck#4142]
In VB6, adding a new component from Project Explorer pops a dialog allowing selection of templates:

![image](https://user-images.githubusercontent.com/25420409/42115250-041ecde2-7bea-11e8-85ae-3cf3845c97c9.png)
The 'Don't show this dialog in the future' option acts per component type, and links to the Tools->Options dialog:
![image](https://user-images.githubusercontent.com/25420409/42115338-56414438-7bea-11e8-9ebf-c2a6dcc81ad7.png)

These settings are held in HKEY_CURRENT_USER\Software\Microsoft\Visual Basic\6.0.
The Templates directory is a simple hierarchy divided by component type. Displayed in the templates dialog is the union of:

1. Blank component (no template)
2. Any files in the corresponding template folder with the standard extension for the component type
3. Any files in the corresponding template folder with an extension of .vbz (wizards)

I believe that when RD is running in VB6, we should make available a similar dialog from CE. Existing folders and registry settings should be consumed such that integration is seamless.

**Comment by bclothier** [comment]
Recall that @ThunderFrame  is working on the PR #4225 -- this is an opportunity to make templates consistent and truly RD's own thing, which I think would help a **LOT** in making the IDE feel a bit more modern. 

Also, I hate dialogs. I'd much rather it be a command over a dialog. Less clicking overall since when you look at the VB6's dialog, it's basically selecting a template from a set from a template which is a lot like selecting from menu, only much less effective since you're now mousing and clicking all over the place.

So my vote is for customized submenu/commands system which can be also extended to handle custom templates for both VBA and VB6.

**Comment by mansellan** [comment]
I agree that direct command access is nicer, but it doesn't scale well. The existing VB6 templating feature picks up code modules from a (configurable) templates folder and imports them as new components. Therefore it's possible for users to set up large collections of templates based on their personal preferences.

There's no way to know how widely known \ used this functionality is, but I think RD should have some way of accommodating the existing workflow. It would be disappointing if users felt the need to keep PE open solely because CE was missing some small part of its functionality.

That said, the existing dialog is horrendously dated. I would prefer to provide a modern equivalent, perhaps similar to the 'Add Item' dialog of modern VS.
